### PR TITLE
ROCm 6.2 updates

### PR DESCRIPTION
**Pull request by kiritigowda**
* GPU Support updates to 6.2+
*  BUILD_WITH_AMD_ADVANCE - removed as no rocdecode available in rocm 6.0 and older
* add rocdecode - #147

### CODE REVIEW

File: docker/rocal-on-ubuntu-20.dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -y install initramfs-tools libnuma-de
         wget ${ROCM_INSTALLER_REPO} && \
         sudo apt-get install -y ./${ROCM_INSTALLER_PACKAGE} && \
         sudo apt-get update -y && \
-        sudo amdgpu-install -y --usecase=graphics,rocm --no-32
+        sudo amdgpu-install -y --usecase=rocm
```
Comment (LakshmiKumar23) [inline docker/rocal-on-ubuntu-20.dockerfile:20]: please add these to the other 2 dockerfiles as well: pytorch  & tensorflow.
Also needs change to the rhel file

File: docker/rocal-on-ubuntu-20-with-pytorch-with-mesa.dockerfile [hunk 2]
```diff
@@ -1,9 +1,6 @@
 ARG PYTORCH_VERSION=latest
```
Comment (LakshmiKumar23) [inline docker/rocal-on-ubuntu-20-with-pytorch-with-mesa.dockerfile:1]: do we still need this file if we are adding rocdecode dependency. This was strictly for graphics drivers

File: docker/rocal-on-ubuntu-20.dockerfile [hunk 3]
```diff
@@ -17,7 +17,7 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -y install initramfs-tools libnuma-de
         wget ${ROCM_INSTALLER_REPO} && \
```
Comment (LakshmiKumar23) [inline docker/rocal-on-ubuntu-20.dockerfile:17]: please remove `--add-architecture` from all dockerfiles

Comment (LakshmiKumar23) [reply docker/rocal-on-ubuntu-20.dockerfile:17]: change line 3 & 4 to rocm6.1 repo.

File: docker/rocal-on-ubuntu-20-with-pytorch-with-mesa.dockerfile (hunk 2 above)
Comment (kiritigowda) [reply docker/rocal-on-ubuntu-20-with-pytorch-with-mesa.dockerfile]: Deleted this file

**Comment by kiritigowda** [comment]
@LakshmiKumar23 -- needs some tests -- might resolve failures

File: docker/rocal-on-ubuntu-20.dockerfile [hunk 4]
```diff
@@ -26,25 +26,15 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -y install build-essential libgtk2.0-
...
 
 # install rocAL neural net dependency
-RUN DEBIAN_FRONTEND=noninteractive apt-get -y install rocblas rocblas-dev miopen-hip miopen-hip-dev migraphx && \
-        mkdir neuralNet && cd neuralNet && wget https://sourceforge.net/projects/half/files/half/1.12.0/half-1.12.0.zip && \
-        unzip half-1.12.0.zip -d half-files && sudo mkdir -p /usr/local/include/half && sudo cp half-files/include/half.hpp /usr/local/include/half && cd
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install half rocblas-dev miopen-hip-dev migraphx-dev
 
 # install rocAL dependency
-RUN DEBIAN_FRONTEND=noninteractive apt-get -y install wget libbz2-dev libssl-dev python-dev python3-dev libgflags-dev libgoogle-glog-dev liblmdb-dev nasm yasm libjsoncpp-dev clang && \
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install rpp-dev wget libbz2-dev libssl-dev python-dev python3-dev libgflags-dev libgoogle-glog-dev liblmdb-dev nasm yasm libjsoncpp-dev clang && \
```
Comment (LakshmiKumar23) [inline docker/rocal-on-ubuntu-20.dockerfile:35]: what about rpp-test package?

File: docker/rocal-with-pytorch.dockerfile [hunk 5]
```diff
@@ -52,7 +42,10 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -y install python3 python3-pip git g+
 
 # install MIVisionX 
 RUN git clone https://github.com/ROCm/MIVisionX.git && cd MIVisionX && \
-        mkdir build && cd build && cmake -DBACKEND=HIP ../ && make -j8 && make install && cd 
+        mkdir build && cd build && cmake -DBACKEND=HIP ../ && make -j8 && make install && cd
```
Comment (LakshmiKumar23) [inline docker/rocal-with-pytorch.dockerfile:45]: can we get mivisionx from package too?

Comment (kiritigowda) [reply docker/rocal-with-pytorch.dockerfile:45]: right now the mivisionx package has no vx_rpp

File: docker/rocal-on-ubuntu-20.dockerfile (hunk 4 above)
Comment (kiritigowda) [reply docker/rocal-on-ubuntu-20.dockerfile:35]: We don't test rpp

**Comment by LakshmiKumar23** [comment]
@SundarRajan28 Please test with this PR for the HIP changes

**Comment by kiritigowda** [comment]
@LakshmiKumar23 -- passing all tests

**Pull request merged**